### PR TITLE
Move "View Assembly Documentation" action out of compiler.js

### PR DESCRIPTION
**Pull request by matsjla**
Depends on #3737 

This is a tiny refactor PR to move the "View Assembly Documentation" Monaco Action out of compiler.js

This makes it so it's possible for multiple panes to re-use the same actions (e.g LLVM IR View)

My plan is to refactor all our custom defined actions into /static/extensions

### CODE REVIEW

File: lib/asm-docs/llvm.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2021, Compiler Explorer Authors
```
Comment (RubenRBS) [inline lib/asm-docs/llvm.ts:1]: 2021

File: static/extensions/view-assembly-documentation.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
+import * as monaco from 'monaco-editor';
```
Comment (RubenRBS) [inline static/extensions/view-assembly-documentation.ts:1]: Needs license

File: tsconfig.frontend.json [hunk 3]
```diff
@@ -9,5 +9,5 @@
     "strictPropertyInitialization": false,
     "lib": ["dom", "es5"]
   },
-  "exclude": ["app.js", "lib", "examples"]
+  "exclude": ["out", "test", "etc", "examples", "app.js", "lib", "examples"]
```
Comment (RubenRBS) [inline tsconfig.frontend.json:12]: This will help with commits check times, nice

**Review by RubenRBS — COMMENTED** [review]
Looking great so far! Nice work

**Comment by RubenRBS** [comment]
Needs a rebase, but they look not to be trivial in one of the files at least :/
Otherwise, the code looks good now that the other PR has been merged

**Comment by matsjla** [comment]
Yeah, I just have a tiny issue with how the decorations are currently stored (because monaco for some reason doesn't keep the decoration deltas on their own data model so we have to store them). Been thinking of ways to make this nicer

**Pull request closed without merging**